conan/cli/commands/remove.py

```diff
@@ -79,18 +80,23 @@ def confirmation(message):
             raise ConanException('--package-query supplied but the pattern does not match packages')
         package_list = conan_api.list.select(ref_pattern, args.package_query, remote)
         multi_package_list = MultiPackagesList()
-        multi_package_list.add("Local Cache" if not remote else remote.name, package_list)
+        multi_package_list.add(cache_name, package_list)
 
+    results_pkg_list = PackagesList() 
+    results = MultiPackagesList()
```

Comment:
You cannot just recreate the lists here, unconditionally, because this discard the information in the above ``multi_package_list``? A lot of tests in https://ci.conan.io/blue/organizations/jenkins/ConanTestSuitev2/detail/PR-14478/2/pipeline/36 are broken, please have a look.

Reply:
the idea was to use `multi_package_list` only for iterating the stuff possible to be removed by the pattern and collect what was confirmed by the user in a separate list.

I'll take a look at the failing tests.

Reply:
Ok, that makes sense, now I see the ``add_xxx`` calls and less failing tests.
I am facing a similar problem in my "upload" branch (I am preparing the same fix for the  upload command), and I have done something similar.
The issue with this is that it is missing the ``package_id`` configuration description. Don't worry about it, it is fine, I will revisit this myself when finishing the upload PR, I might change the API a bit for ``PackageLists`` to help with this.